File: app/src/main/java/com/zerostress/manager/models/VoiceChatMessage.java
package com.zerostress.manager.models;

public class VoiceChatMessage {
    private String id;
    private String senderId;
    private String senderName;
    private String text;
    private long timestamp;

    public VoiceChatMessage() {}

    public VoiceChatMessage(String senderId, String senderName, String text) {
        this.senderId = senderId;
        this.senderName = senderName;
        this.text = text;
        this.timestamp = System.currentTimeMillis();
    }

    // Getters and Setters
    public String getId() { return id; }
    public void setId(String id) { this.id = id; }

    public String getSenderId() { return senderId; }
    public void setSenderId(String senderId) { this.senderId = senderId; }

    public String getSenderName() { return senderName; }
    public void setSenderName(String senderName) { this.senderName = senderName; }

    public String getText() { return text; }
    public void setText(String text) { this.text = text; }

    public long getTimestamp() { return timestamp; }
    public void setTimestamp(long timestamp) { this.timestamp = timestamp; }
}

File: app/src/main/java/com/zerostress/manager/models/VoiceChatMessage.kt
package com.zerostress.manager.models

import com.google.firebase.firestore.IgnoreExtraProperties

@IgnoreExtraProperties
data class VoiceChatMessage(
    var id: String = "",
    var senderId: String = "",
    var senderName: String = "",
    var text: String = "",
    var timestamp: Long = 0L
) {
    companion object {
        fun create(senderId: String, senderName: String, text: String): VoiceChatMessage =
            VoiceChatMessage(
                senderId = senderId,
                senderName = senderName,
                text = text,
                timestamp = System.currentTimeMillis()
            )
    }
}
